rinex-cli/src/positioning/interp/orbit.rs

```diff
@@ -0,0 +1,398 @@
+use crate::cli::Context;
+use gnss_rtk::prelude::{
+    AprioriPosition, Epoch, InterpolationResult as RTKInterpolationResult, SV,
+};
+use rinex::navigation::Ephemeris;
+use std::collections::HashMap;
+
+use super::Buffer as BufferTrait;
+
+struct Buffer {
+    order: usize,
+    inner: Vec<(Epoch, (f64, f64, f64))>,
+}
+
+impl BufferTrait<(f64, f64, f64)> for Buffer {
+    fn malloc(order: usize) -> Self {
+        if order % 2 == 0 {
+            panic!("only odd interpolation orders currently supported");
+        }
+        Self {
+            order,
+            inner: Vec::with_capacity(order + 1),
+        }
+    }
+    fn len(&self) -> usize {
+        self.inner.len()
+    }
+    /// Fill internal Buffer, which does not tolerate data gaps
+    fn at_capacity(&self) -> bool {
+        self.inner.len() == self.order + 1
```

Comment:
Ooh, this makes more sense than comparing against 3 like the previous version did :upside_down_face:.

Reply:
There are two dfiferent implementations of the Trait at the moment.
Orbital implementation is where this may vary

Reply:
No, there are two identical implementations of the trait:

![image](https://github.com/georust/rinex/assets/308347/655b8e5f-9825-47aa-abcf-610048af9da5)

You could do:

```rust
struct RingBuffer<T> {
    inner: Box<[T]>, // or Vec<T>
    head: usize,
    tail: usize,
}

type TimeBuffer = RingBuffer<(Epoch, f64)>;
type PositionBuffer = RingBuffer<(Epoch, (f64, f64, f64))>;
```